### PR TITLE
Setup Testing Utilities and Examples

### PR DESCRIPTION
**Pull request by Austio**
Pivotal: https://www.pivotaltracker.com/story/show/139238377

This PR is part 1 of a few but can be merged now.  Will readd the slots for images/icons in next pr.

##### Structure
 - Creates a components and example folder in src
 - For Components, adds a folder and index export per component
 - For Example, hosts the example app we will use
 - Runs lint as part of test command

##### Utilities Methods
 - Mounting a component for testing
 - Triggering Events
 - Inputing Text

##### Component
Sets testing examples for methods on TextInput component.  I split the events so that they each handle their own function, this will allow us to move this input logic into a mixin in the future and import that into all of our input based functions and then just change the things that are different with specific inputs.

### CODE REVIEW

File: src/components/text-input/index.vue [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        currentValue: this.value,
+      };
+    },
+    props: {
+      type: {
+        type: String,
+        default: 'text',
+      },
+      class: {
+        default: '',
```
Comment (Austio) [inline src/components/text-input/index.vue]: Would be pretty simple here to support some straight forward classes that we could use for input states and then merge all of that into a classObject with whatever is passed in as props.

{ 
  vg-input-focus: isFocused(), 
  vg-input-error...
}

File: src/components/text-input/index.vue [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      },
+      handleFocus(event) {
+        this.$emit('focus', event);
+      },
+      handleBlur(event) {
+        this.$emit('blur', event);
+      },
+      handleInput(event) {
+        this.currentValue = event.target.value;
+        this.$emit('input', currentValue);
```
Comment (Austio) [inline src/components/text-input/index.vue]: emitting an `input` and accepting `value` as props allows people to use the inputs using v-model

File: src/examples/index.js [hunk 3]
```diff
@@ -1,8 +1,8 @@
-/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable */
```
Comment (aurelienbottazini) [inline src/examples/index.js]: can we disable just the specific line that is causing the problem?

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import Components from './components';
+
+// Converts pascalCased to pascal-cased
+function genomeName(key) {
```
Comment (aurelienbottazini) [inline src/index.js]: 💡  remove comment and rename to 
`convertPascalToKebabCase`?

File: test/unit/util.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
+/* eslint-disable */
```
Comment (aurelienbottazini) [inline test/unit/util.js]: ❓ disable just the problematic line ❓

File: src/examples/index.js (hunk 3 above)
Comment (Austio) [inline src/examples/index.js]: We are not actually using anything in this file that is distributed, it is just an example testbed where we can add things and make requests for testing locally.

File: src/index.js (hunk 4 above)
Comment (Austio) [inline src/index.js]: This is super small so don't want to bikeshed too much on it, but my thought is that genomeName describes the intent of the method whereas convertPascalToKebab describes how it does it right now and would actually need to be `appendVgAndConvertPascalToKebab`.  

In scope of this it is not a big deal but if we ever changed what the genomeName would import the components (say give options to import as VgInputText or vg-input-text or remove the vg-) we would need to change the function name, how it is referenced in the install function and its internals.  

I guess technically we would have to change internals and comment this way, so maybe just remove comment?

File: test/unit/util.js (hunk 5 above)
Comment (Austio) [inline test/unit/util.js]: These are utils that we will be using to run tests, don't think the linting is valuable here due to that.  (note i started with adding single lines but there were too many)

File: src/index.js (hunk 4 above)
Comment (aurelienbottazini) [inline src/index.js]: 👍 ok for just removing the comment

File: test/unit/util.js (hunk 5 above)
Comment (aurelienbottazini) [inline test/unit/util.js]: oki

Comment (aurelienbottazini) [inline test/unit/util.js]: 👍

File: src/examples/index.js (hunk 3 above)
Comment (aurelienbottazini) [inline src/examples/index.js]: 👍

**Pull request merged**